### DISCUSSION TITLE
dockerhub 中的master镜像无法启动

### DISCUSSION THREAD

**Discussion by xxoxx** [hhyo/Archery#2507] [Q&A]
### 重现步骤

我看到releases中最新的版本为v1.10.0， 因此我就通过docker composer的方式 运行了v1.10.0版本的镜像，经过测试，一切也正常

后续我在本项目的master提交记录中，看到修复了很多相关的bug, 因此我也看到github的actions会自动打包master并上传到dockerhub镜像，   因此我将 docker-compose.yml中的镜像版本由 hhyo/archery:v1.10.0  更换为hhyo/archery:master，  但是更改后发现不能成功连接mysql，   我尝试将v1.10.0的settings.py来替换， 但是替换后发现archery容器无法启动

### 预期外的结果

希望修改bug后的版本能够及时同步到dockerhub中， 并能够正常启动

### 版本

master

### 部署方式

Docker

**Comment by LeoQuote** [comment]
报错，报啥错呢哥？无法启动，咋个无法启动呢？顺手贴一下报错信息呗

**Comment by xxoxx** [comment]
<img width="1439" alt="图片" src="https://github.com/hhyo/Archery/assets/8625218/f2d05bab-0327-471a-b870-c9d6540b4d9f">
@LeoQuote     哥，报这个错，  明显应该是Mysql的数据库连接信息没有对，   我的复现步骤为     下载官方master分支代码，进入src/docker-compose 目录，  将docker-compose.yml文件中的 image: hhyo/archery:v1.10.0  修改为了  image: hhyo/archery:master 然后就这样子了， 


为什么要把镜像版本改为image: hhyo/archery:master，  我是看github的提交记录中， 在v1.10.0这个版本后， 在master分支上最近修改了很多bug， 麻烦你抽空看一下呢

**Reply by LeoQuote** [reply]
你的问题吧，你看一下你的镜像还是 python3.8 这都什么时候的版本了

**Comment by xxoxx** [comment]
@LeoQuote    两个问题， 
1. docker hub中的镜像版本为：hhyo/archery:master   这个标签一直在被覆盖， 正常使用呢，没什么问题， 但是如果在国内的docker服务器上配置了加速镜像后， 比如我这配置的阿里的加速镜像仓库， 它只会保存它当时拉的第一个， 后续这个标签如果发生了更新的话，它就不会重新拉取，  这就是拉到的hhyo/archery:master这个版本的python3.8版本的原因， 后续我跳过加速仓库，然后就拉到了正确的镜像，但是国内目前的网络环境，不使用加速仓库拉镜像的话，是非常慢的。 建议： 每次修改bug后，还是维护一下版本号，比如 v1.10.0.10 等， 然后docker hub中的标签跟这个版本号同步，  往docker hub推着镜像的时候建议别覆盖。
2. 当我拉取正确的镜像后，运行docker-compose -f docker-compose.yml up -d   提示：Error response from daemon: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "dockerize": executable file not found in $PATH: unknown， 应该是hhyo/archery:master的镜像中没有dockerize这个，或者它没在环境变量中，麻烦检查一下，   我后续把docker-compose.yml文件中的 entrypoint: "dockerize -wait tcp://mysql:3306 -wait tcp://redist:6379 -timeout 60s /opt/archery/src/docker/startup.sh"  这一段注释了，然后就能正常启动了 也不知道这段注释后是否有什么影响。

以上为个人建议，请酌情考虑，谢谢。

**Reply by LeoQuote** [reply]
发版问题，这边是懒得发那么多版的，你完全可以自己clone 代码，使用国内的pypi 镜像自己进行镜像构建，也不慢。

dockerize 的相关代码已经清理 ，参考 https://github.com/hhyo/Archery/commit/d8159f4fa5f61ea1f67d078602c311971ae23817 ，可以使用一下最新的 docker compose 代码

**Comment by xxoxx** [comment]
![图片](https://github.com/hhyo/Archery/assets/8625218/be2d261c-518a-40d9-9d73-fd3bda0347ad)
